src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,608 @@ const ActorInit En_Box_InitVars = {
+            EnBox_SetupAction(this, EnBox_Open);
+        }
+    }
+}
+
+/**
+ * Plays an animation to its end, playing sounds at key points
+ */
+void EnBox_Open(EnBox* this, GlobalContext* globalCtx) {
+    u16 sfxId;
+
+    this->dyna.actor.flags &= ~0x80;
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelanime)) {
+        if (this->unk_1F4 > 0) {
+            if (this->unk_1F4 < 120) {
+                this->unk_1F4++;
+            } else {
+                Math_ApproxF(&this->unk_1B0, 0.0f, 0.05f);
+            }
+        } else {
+            if (this->unk_1F4 > -120) {
+                this->unk_1F4--;
+            } else {
+                Math_ApproxF(&this->unk_1B0, 0.0f, 0.05f);
+            }
+        }
+    } else {
+        sfxId = 0;
+
+        if (func_800A56C8(&this->skelanime, 30.0f)) {
+            sfxId = NA_SE_EV_TBOX_UNLOCK;
+        } else if (func_800A56C8(&this->skelanime, 90.0f)) {
+            sfxId = NA_SE_EV_TBOX_OPEN;
+        }
+
+        if (sfxId) {
+            Audio_PlaySoundGeneral(sfxId, &this->dyna.actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+
+        if (this->skelanime.limbDrawTbl[3].z > 0) {
+            this->unk_1B0 = (0x7D00 - this->skelanime.limbDrawTbl[3].z) * 0.00006f;
+            if (this->unk_1B0 < 0.0f) {
+                this->unk_1B0 = 0.0f;
+            } else if (this->unk_1B0 > 1.0f) {
+                this->unk_1B0 = 1.0f;
+            }
+        }
+    }
+}
+
+void func_809C9EF8(EnBox* this, GlobalContext* globalCtx) {
+    Vec3f sp4c_3f;
+    Vec3f sp40_3f;
+    Vec3f sp34_3f;
+    f32 f0;
+
+    sp40_3f = D_809CA814;
+    sp34_3f = D_809CA820;
+    this->iceEffTimer++;
+    func_8002F974(&this->dyna.actor, 0x31F1);
+    if (Math_Rand_ZeroOne() < 0.3f) {
+        f0 = Math_Rand_ZeroOne();
+        sp4c_3f = this->dyna.actor.posRot.pos;
+        f0 = f0 + f0 - 1.0f;
+        if (this->type == 5 || this->type == 6 || this->type == 7 || this->type == 8) {
+            sp4c_3f.x += f0 * 10.0f * Math_Sins(this->dyna.actor.posRot.rot.y + 0x4000);
+            sp4c_3f.z += f0 * 10.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x4000);
+            f0 = Math_Rand_ZeroOne();
+            f0 = f0 + f0 - 1.0f;
+            sp40_3f.x = f0 * 0.8f * Math_Sins(this->dyna.actor.posRot.rot.y);
+            sp40_3f.y = 1.8f;
+            sp40_3f.z = f0 * 0.8f * Math_Coss(this->dyna.actor.posRot.rot.y);
+        } else {
+            sp4c_3f.x += f0 * 20.0f * Math_Sins(this->dyna.actor.posRot.rot.y + 0x4000);
+            sp4c_3f.z += f0 * 20.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x4000);
+            f0 = Math_Rand_ZeroOne();
+            f0 = f0 + f0 - 1.0f;
+            sp40_3f.x = f0 * 1.6f * Math_Sins(this->dyna.actor.posRot.rot.y);
+            sp40_3f.y = 1.8f;
+            sp40_3f.z = f0 * 1.6f * Math_Coss(this->dyna.actor.posRot.rot.y);
+        }
+        func_8002AA44(globalCtx, &sp4c_3f, &sp40_3f, &sp34_3f, 0x96);
```

Comment:
heres the prototype if you want to name these proper. if not id at least remove 3f from the name

`void EffectSsIceSmoke_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale)`

scale in deimal would be appreciated